response/max_age.go

```diff
@@ -18,6 +21,41 @@ func maxAge(ctx context.Context, uri string, cfg *config.Config) int {
 		return int(cfg.CacheTimeLong.Seconds())
 	}
 
+	pagePath := getPagePath(ctx, uri)
+
+	releaseTime, statusCode, err := getReleaseTime(pagePath, cfg.LegacyCacheAPIURL)
+
+	if err != nil {
+		log.Error(ctx, "error calculating the max-age directive", err)
+		return int(cfg.CacheTimeErrored.Seconds())
```

Comment:
I can look into this, but how would this look like? We need to pass the configuration all the way down from `service.go` because we need the following:
- Babbage URL
- Legacy Cache API URL
- The 4 configurable cache times
- The publish expiry offset

I can only think of a couple different implementations, and none of them are great:
1. We calculate the integer value of the 4 configurable cache times and the publish expiry offset, and add it to the `Config` struct. This will look strange because we will have a set of properties that we don't need to read from the environment variables, and are essentially repeated.
2. We create a new, different `Config` struct (maybe we can call it `Settings`), and this will contain everything I mentioned in the first paragraph (the 2 URLs and the integer values of the 4 cache times and the publish expiry offset). So everything that `Config` has, except for the bind address, the graceful shutdown timeout, the health check variables and the OpenTelemetry ones. We create this `Settings` struct once during the proxy setup, and pass it down to `proxy.manage` instead of passing `Config`. This would look a bit better, but it comes with a cost: in the code we're currently using the cache times as `time.Time` for time calculations, so if they now come as integers, we would have to make the conversion back to time again, which would defeat the purpose.

Unless you know of a tidy way of doing this in Go, I would suggest leaving it as it is, because I am not sure that this is an actual problem. Apart from the conversion in line 21, the rest of them happen after we make a call to the API. So whatever efficiency gains we get, they're going to be nothing compared to the cost of making a network call. In addition to that, we're going to do some performance tests next week, so we could test the difference between this implementation and an alternative implementation in which we don't make any conversions. And then, depending on the results of the tests, we can make the changes or leave it as it is.